apiserver/facades/client/charms/client.go

```diff
@@ -570,6 +594,18 @@ func (a *API) resolveOneCharm(arg params.ResolveCharmWithChannel, mac *macaroon.
 	return result
 }
 
+func validateOrigin(origin params.CharmOrigin, schema string) error {
+	if (corecharm.Local.Matches(origin.Source) && !charm.Local.Matches(schema)) ||
+		(corecharm.CharmStore.Matches(origin.Source) && !charm.CharmStore.Matches(schema)) ||
+		(corecharm.CharmHub.Matches(origin.Source) && !charm.CharmHub.Matches(schema)) {
+		return errors.NotValidf("origin source %q with schema", origin.Source)
+	}
+	if corecharm.CharmHub.Matches(origin.Source) && origin.Architecture == "" {
```

Comment:
I thought, due to the migration bug https://bugs.launchpad.net/juju/+bug/1911800, architecture would be required with a CS charm too.

Reply:
Well due to that bug, I think we should undo this, but as a separate PR and tackle that holistically.